staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go

```diff
@@ -254,19 +283,136 @@ type ClaimMappings struct {
+	// If groups.claim is set, the prefix must be specified (and can be the empty string).
 	// +optional
 	Groups PrefixedClaimOrExpression `json:"groups,omitempty"`
+
+	// uid represents an option for the uid attribute.
+	// Claim must be a singular string claim.
+	// Expression must produce a string value.
+	// +optional
+	UID ClaimOrExpression `json:"uid,omitempty"`
+
+	// extra represents an option for the extra attribute.
+	// If multiple mappings have the same key, the result will be a concatenation of all values
+	// with the order preserved.
+	// If the value is empty, the extra mapping will not be present.
+	//
+	// hard-coded extra key/value
+	// - key: "foo"
+	//   valueExpression: "'bar'"
+	// This will result in an extra attribute - foo: ["bar"]
+	//
+	// hard-coded key, value copying claim value
+	// - key: "foo"
+	//   valueExpression: "claims.some_claim"
+	// This will result in an extra attribute - foo: [value of some_claim]
+	//
+	// hard-coded key, value derived from claim value
+	// - key: "admin"
+	//   valueExpression: '(has(claims.is_admin) && claims.is_admin) ? "true":""'
+	// This will result in:
+	//  - if is_admin claim is present and true, extra attribute - admin: ["true"]
+	//  - if is_admin claim is present and false or is_admin claim is not present, no extra attribute will be added
+	//
+	// +optional
+	Extra []ExtraMapping `json:"extra,omitempty"`
 }
 
 // PrefixedClaimOrExpression provides the configuration for a single prefixed claim or expression.
 type PrefixedClaimOrExpression struct {
 	// claim is the JWT claim to use.
+	// Mutually exclusive with expression.
 	// +optional
 	Claim string `json:"claim"`
 	// prefix is prepended to claim's value to prevent clashes with existing names.
-	// +required
+	// prefix needs to be set if claim is set and can be the empty string.
+	// Mutually exclusive with expression.
+	// +optional
 	Prefix *string `json:"prefix"`
+
+	// expression represents the expression which will be evaluated by CEL.
+	// Must produce a string.
+	//
+	// CEL expressions have access to the contents of the token claims, organized into CEL variable:
+	// - 'claims' is a map of claim names to claim values.
+	//   For example, a variable named 'sub' can be accessed as 'claims.sub'.
+	//   Nested claims can be accessed using dot notation, e.g. 'claims.email.verified'.
+	//
+	// Documentation on CEL: https://kubernetes.io/docs/reference/using-api/cel/
+	//
+	// Mutually exclusive with claim and prefix.
+	// +optional
+	Expression *string `json:"expression"`
+}
+
+// ClaimOrExpression provides the configuration for a single claim or expression.
+type ClaimOrExpression struct {
+	// claim is the JWT claim to use.
+	// Either claim or expression must be set.
+	// Mutually exclusive with expression.
+	// +optional
+	Claim string `json:"claim"`
+
+	// expression represents the expression which will be evaluated by CEL.
+	// Must produce a string.
```

Comment:
In case of UID and Username claim mappings, `null` will break things. For these 2 the expected type is string. For groups, we handle `null`, `[]string` and `string`.